AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -648,10 +643,15 @@ - (BOOL)shouldMeasureWithSizeRange:(ASSizeRange)constrainedSize
     }
   }
   
-  // only calculate the size if
-  //  - we haven't already
-  //  - the constrained size range is different
-  return (!_flags.isMeasured || !ASSizeRangeEqualToSizeRange(constrainedSize, _constrainedSize));
+  // Only generate a new layout if:
+  // - The current layout is dirty
+  // - The passed constrained size is different than the layout's constrained size
+  return ([self _dirtyLayout] || !ASSizeRangeEqualToSizeRange(constrainedSize, _layout.constrainedSizeRange));
```

Comment:
This should be called _layoutIsDirty or something else; it reads like an action / verb currently